components/ai_chat/core/browser/utils.cc

```diff
@@ -178,6 +182,56 @@ const std::string& GetActionTypeQuestion(mojom::ActionType action_type) {
   return iter->second;
 }
 
+uint32_t GetMaxPageContentLengthFromContextSize(
+    std::optional<uint32_t> context_size) {
+  return static_cast<uint32_t>(
+      context_size.value_or(kCustomModelDefaultContextSize) *
+      kDefaultCharsPerToken);
+}
+
+uint32_t GetMaxPageContentLengthFromModel(const mojom::Model& model) {
+  // For Leo models, simply return the max_page_content_length.
+  if (model.options->is_leo_model_options()) {
+    return model.options->get_leo_model_options()->max_page_content_length;
+  }
+
+  // For custom models, calculate the max length based on the context size.
+  return GetMaxPageContentLengthFromContextSize(
+      model.options->get_custom_model_options()->context_size);
+}
+
+uint32_t GetLongConversationWarningLengthFromContextSize(
+    std::optional<uint32_t> context_size) {
+  return static_cast<uint32_t>(
+      context_size.value_or(kCustomModelDefaultContextSize) *
+      kDefaultCharsPerToken * kMaxContentLengthThreshold);
```

Comment:
This multiplication doesn't look safe. It looks like we should be using `base::CheckedNumeric` to make sure we are not overflowing here.